### PR TITLE
createTag.sh

### PR DESCRIPTION
**Pull request by fgalan**
Tested in the 'files' mode ('push' mode will be tested next time we need a tag).

Files seems to be correctly modified. After running script on release/1.2.0 branch git diff shows the following:

```
fermin@trantor:~/src/fiware-orion$ git diff
diff --git a/CHANGES_NEXT_RELEASE b/CHANGES_NEXT_RELEASE
index f8aec1c..e69de29 100644
--- a/CHANGES_NEXT_RELEASE
+++ b/CHANGES_NEXT_RELEASE
@@ -1,6 +0,0 @@
-- Fix: UTC as time in log file (Issue #2232)
-- Fix: Removed capping in NGSIv2 update forwarding (Issue #2193)
-- Fix: In some cases, compound attribute values were forwarded as empty strings (Issue #2237)
-- Fix: attribute overwritten when JSON value used in PUT /v2/entities/<eId>/attr/<attrName>/values (Issues #2246 and #2248)
-- Fix: subscription equal filter evaluation on update context logic (#2222)
-- Fix: capturing invalid id patterns at json parse time for v2 subscriptions - this bug made the broker crash when receiving invalid id-patterns (Issue #2257)
diff --git a/docker/Dockerfile b/docker/Dockerfile
index 59f534e..67a0c84 100644
--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -3,7 +3,7 @@ FROM centos:centos6.6
 MAINTAINER FIWARE Orion Context Broker Team. Telefónica I+D

 ENV ORION_USER orion
-ENV GIT_REV_ORION 1.2.0
+ENV GIT_REV_ORION 1.2.1
 ENV CLEAN_DEV_TOOLS 1

 RUN adduser --comment "${ORION_USER}" ${ORION_USER}
diff --git a/rpm/SPECS/contextBroker.spec b/rpm/SPECS/contextBroker.spec
index e489f73..57743dc 100644
--- a/rpm/SPECS/contextBroker.spec
+++ b/rpm/SPECS/contextBroker.spec
@@ -164,6 +164,14 @@ if [ "$1" == "0" ]; then
 fi

 %changelog
+* Fri Jun 10 2016 Fermin Galan <fermin.galanmarquez@telefonica.com> 1.2.1-1
+- Fix: UTC as time in log file (Issue #2232)
+- Fix: Removed capping in NGSIv2 update forwarding (Issue #2193)
+- Fix: In some cases, compound attribute values were forwarded as empty strings (Issue #2237)
+- Fix: attribute overwritten when JSON value used in PUT /v2/entities/<eId>/attr/<attrName>/values (Issues #2246 and #2248)
+- Fix: subscription equal filter evaluation on update context logic (#2222)
+- Fix: capturing invalid id patterns at json parse time for v2 subscriptions - this bug made the broker crash when receiving invalid id-patterns (Issue #2257)
+
 * Thu Jun 02 2016 Fermin Galan <fermin.galanmarquez@telefonica.com> 1.2.0-1
 - Add: notification attributes blacklist for NGSIv2 notifications (#2064)
 - Add: NGSIv2 custom notifications (#2015)
diff --git a/src/app/contextBroker/version.h b/src/app/contextBroker/version.h
index c6bd698..3b14d75 100644
--- a/src/app/contextBroker/version.h
+++ b/src/app/contextBroker/version.h
@@ -28,6 +28,6 @@



-#define ORION_VERSION "1.2.0"
+#define ORION_VERSION "1.2.1"

 #endif  // SRC_APP_CONTEXTBROKER_VERSION_H_
```

### CODE REVIEW

**Comment by fgalan** [comment]
Please don't delete branch after merge (it will be used for a twin PR to develop.

File: scripts/build/createTag.sh [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+files=off
+commit=off
+tag=off
+push=off
+
+changelog=CHANGES_NEXT_RELEASE
+
+mode=""
+
+while [ "$#" != 0 ]
```
Comment (dmoranj) [inline scripts/build/createTag.sh]: All this management of command line arguments could be handled using [getopts](http://man.cx/getopts(1)).

File: scripts/build/createTag.sh [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# FIXME: this script assumes that it is run from the fiware-orion repository run
```
Comment (kzangeli) [inline scripts/build/createTag.sh]: If you want to, you can check that this is true.
I did this in some script ...
I think I simply checked that .git/config exists and is a readable file

File: scripts/build/createTag.sh [hunk 3]
```diff
@@ -0,0 +1,239 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (kzangeli) [inline scripts/build/createTag.sh]: 2016

File: scripts/build/createTag.sh [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+grep "ENV GIT_REV_ORION $currentTag" $PROJECT_DIR/docker/Dockerfile
+if [ "$?" != "0" ]; then
+  echo $0: GIT_REV_ORION does not use current tag in Dockerfile
+  exit 1
+fi
+
+# Calculate next tag
+Z=$((Z+1))
+nextTag=$X.$Y.$Z
+echo $0: next tag is: "'"${nextTag}"'"
```
Comment (kzangeli) [inline scripts/build/createTag.sh]: leftover debug message?

File: scripts/build/createTag.sh [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [[ $branch != release/* ]]; then
+  echo $0: you are not in a release branch but in "'"${branch}"'"
+  exit 1
+fi
+
+
+# Get current tag
+currentTag=$(grep "define ORION_VERSION" $PROJECT_DIR/src/app/contextBroker/version.h  | awk -F\" '{ print $2 }')
+echo $0: current tag is: "'"${currentTag}"'"
```
Comment (kzangeli) [inline scripts/build/createTag.sh]: leftover?

Comment (fgalan) [inline scripts/build/createTag.sh]: No, not really. Part of the normal output of the script, in order to report user how the things are going.

File: scripts/build/createTag.sh (hunk 4 above)
Comment (fgalan) [inline scripts/build/createTag.sh]: No, not really. Part of the normal output of the script, in order to report user how the things are going.

File: scripts/build/createTag.sh (hunk 1 above)
Comment (fgalan) [inline scripts/build/createTag.sh]: Fixed in 0cfbfe7.

At the end, I changed to use short options (`-m`, `-u`, `-c`) as it seems that the alternative for long options (`getopt`, without `s`) is not portable.

File: scripts/build/createTag.sh (hunk 2 above)
Comment (fgalan) [inline scripts/build/createTag.sh]: Fixed in 0cfbfe7

File: scripts/build/createTag.sh (hunk 3 above)
Comment (fgalan) [inline scripts/build/createTag.sh]: Fixed in 0cfbfe7

File: scripts/build/createTag.sh (hunk 4 above)
Comment (fgalan) [inline scripts/build/createTag.sh]: NTC (I guess)

File: scripts/build/createTag.sh (hunk 5 above)
Comment (fgalan) [inline scripts/build/createTag.sh]: NTC (I guess)

**Comment by dmoranj** [comment]
LGTM

File: scripts/build/createTag.sh (hunk 4 above)
Comment (kzangeli) [inline scripts/build/createTag.sh]: Yes, NTC

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**